docs/sql-data-sources-parquet.md

```diff
@@ -252,6 +252,71 @@ REFRESH TABLE my_table;
 
 </div>
 
+## Columnar Encryption
+
+
+Since Spark 3.2, columnar encryption is supported for Parquet tables with Apache Parquet 1.12+.
+
+Parquet uses the envelope encryption practice, where file parts are encrypted with "data encryption keys" (DEKs), and the DEKs are encrypted with "master encryption keys" (MEKs). The DEKs are randomly generated by Parquet for each encrypted file/column. The MEKs are generated, stored and managed in a Key Management Service (KMS) of user’s choice. Parquet maven [repository]( https://repo1.maven.org/maven2/org/apache/parquet/parquet-hadoop/1.12.0/) has a jar with a mock KMS implementation that allows to run column encryption and decryption using a spark-shell only, without deploying a KMS server (download the `parquet-hadoop-tests.jar` file and place it in the Spark `jars` folder):
```

Comment:
Parquet maven -> The Parquet Maven
(Fix the hyperlink - space in parens)
But do we need to talk about JARs here? the idea is just that you need to include this as a dependency one way or the other

Reply:
> Parquet maven -> The Parquet Maven
> (Fix the hyperlink - space in parens)

Will do.

> But do we need to talk about JARs here? the idea is just that you need to include this as a dependency one way or the other

This is a special KMS client class, used only for sample illustration and initial demonstrations of Parquet encryption feature (not in production deployment), so the Spark distribution doesn't include it. Still, this class makes it easier to get started with this feature, and having a direct link to its jar would save some overhead for the developers.

Reply:
I know it's not in Spark, but, why repeat 'how to include a JAR' here?

Reply:
Having a link would make it easier to find the file. But we can indeed skip the "find the jar" instructions, and just mention a need to include the parquet-hadoop-tests v1.12.0 dependency.